luda-adventure/src/app/game/types/game_object/types/collider.rs

```diff
@@ -0,0 +1,123 @@
+use crate::app::game::Tile;
+use namui::prelude::*;
+
+pub type CollisionBox = Rect<Tile>;
+
+#[derive(ecs_macro::Component)]
+pub struct Collider {
+    collision_offset_rect: Rect<Tile>,
+}
+
+impl Collider {
+    pub fn new(collision_offset_rect: Rect<Tile>) -> Self {
+        Self {
+            collision_offset_rect,
+        }
+    }
+    pub fn get_collision_box(&self, xy: Xy<Tile>) -> CollisionBox {
+        Rect::Xywh {
+            x: xy.x + self.collision_offset_rect.x(),
+            y: xy.y + self.collision_offset_rect.y(),
+            width: self.collision_offset_rect.width(),
+            height: self.collision_offset_rect.height(),
+        }
+    }
+}
+
+pub fn simplify_collision_box_list(collision_box_list: Vec<CollisionBox>) -> Vec<CollisionBox> {
+    let mut simplified_collision_box_list = Vec::new();
+    for collision_box in collision_box_list {
+        merge_or_create_collision_box(collision_box, &mut simplified_collision_box_list);
+    }
+    simplified_collision_box_list
+}
+fn merge_or_create_collision_box(
+    collision_box: CollisionBox,
+    simplified_collision_box_list: &mut Vec<CollisionBox>,
+) {
+    let collision_box_area = get_collision_box_area(collision_box);
+    for simplified_collision_box in simplified_collision_box_list.iter_mut() {
+        let merged_collision_box =
+            simplified_collision_box.get_minimum_rectangle_containing(collision_box);
+        let merged_area = get_collision_box_area(merged_collision_box);
+        let summed_area = collision_box_area + get_collision_box_area(*simplified_collision_box);
+        let can_merge = merged_area == summed_area;
```

Comment:
This mean you don't merge ㄱ shaped boxes, right?

Reply:
Yes, actually this shape.
```
■
  ■
```
Outdated